src/opensearch_sdk_py/server/async_extension_host.py

```diff
@@ -54,24 +58,17 @@ def on_input(self, input: StreamInput) -> StreamOutput:
                 error_handler = ActionNotFoundRequestErrorHandler(self.extension, request)
                 output = error_handler.handle(request, input)
         else:
-            response = OutboundMessageResponse().read_from(input, header)
-            # TODO: Error handling
+            response: OutboundMessageResponse = OutboundMessageResponse().read_from(input, header)
             if response.is_error:
+                # TODO: Error handling
                 output = None
                 logging.warning(f"< error {header}")
             else:
-                ack_response = AcknowledgedResponse().read_from(input)
-                logging.debug(f"< response {response}, {ack_response}")
-                output = StreamOutput()
-                response = OutboundMessageResponse(
-                    response.thread_context_struct,
-                    response.features,
-                    InitializeExtensionResponse(self.extension.name, self.extension.implemented_interfaces),
-                    response.version,
-                    self.extension.init_response_request_id,
-                    response.is_handshake,
-                    response.is_compress,
-                )
-                response.write_to(output)
-                logging.info(f"> {response}")
+                logging.info(f"< response {response}")
+                if response.request_id in self.response_handlers:
```

Comment:
Could probably move this into `self.response_handlers.handle`, it would just return `None` if unhandled.

Reply:
I wanted to keep the implementation similar to the request handling above it.  I agree we could move both into the handlers.